magento/pwa-studio
# fix: product/category root component prefetch

## Description

The Category/PDP had some "magic" wherein we prefetch the product data for the rendered products up front and then used a fragment matcher to manually pull from the cache on the PDP. To do this we made a "big" query up front for category + product data. 

This PR moves the prefetch logic to the gallery items which makes a prefetch query when mounted. This results in a smaller initial category request but N more requests for product data per gallery item.

I also updated breadcrumbs to follow the new pattern and to use cache-and-network.

## Related Issue





Closes [PWA-778](https://jira.corp.magento.com/browse/PWA-778)
## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


1. Go to the category page. Ensure that you can see the product images/prices/titles and that you can click to go to the product page.
2. Go to the product page. Ensure that nothing is broken.
2. Check network requests - you should see one for product details and one for breadcrumbs (this is the "network" part from "cache-and-network").

## Screenshots / Screen Captures (if appropriate)
[![Image from Gyazo](https://i.gyazo.com/42ed6dc88fc185b6103e659e0b773376.gif)](https://gyazo.com/42ed6dc88fc185b6103e659e0b773376)

## Checklist

* I have updated the documentation accordingly, if necessary.
